File: Java/twoSum.java
// one pass HashMap
class Solution {
    public int[] twoSum(int[] nums, int target) {
        Map<Integer, Integer> numHashMap = new HashMap<>();
        // HashMap => val : index 
        
        int l =nums.length;
        int i;
        
        // loop on the input array 
        for ( i = 0; i < l ; i++) {
            // find the complement that adds up to the target
            int comp = target - nums[i];

            // check if it is in the hashMap
            if (numHashMap.containsKey(comp)) {
                return new int[]{numHashMap.get(comp), i};
            }
            // in none, add it to the HashMap
            numHashMap.put(nums[i], i);       
        }
        // return empty integer array if there's no solution
        return new int[]{};

    }
}        

File: python/Two_Sum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        previousMap = {} # val : index

        for i, n in enumerate(nums):
            # check if the diff equals target minus n
            diff = target - n

            # if its in the HashMap the return a pairs of the indexes
            if diff in previousMap:
                return [previousMap[diff], i]

            # if we don't find the solution we are going to update the HashMap
            previousMap[n] = i
